internal/pkg/composable/providers/kubernetessecrets/kubernetes_secrets.go

```diff
@@ -175,11 +176,13 @@ func (p *contextProviderK8SSecrets) Fetch(key string) (string, bool) {
 	p.store.AddConditionally(key, sd, true, func(existing secret, exists bool) bool {
 		if !exists {
 			// no existing secret in the cache thus add it
+			p.logger.Info("Fetch: \"", key, "\" inserted. Resource Version of secret: \"", apiSecretResourceVersion, "\"")
```

Comment:
It would be nicer to use a format string instead I think
```suggestion
			p.logger.Infof("Fetch: \"%s\" inserted. Resource Version of secret: \"%s"\"", key, apiSecretResourceVersion)
```

Same for a few other log lines added in this PR.

Reply:
No problem changing into Infof but since everything we need to log is already a string, I don't see the point of paying the extra price of calling [fmt.Sprintf](https://github.com/uber-go/zap/blob/6d482535bdd97f4d97b2f9573ac308f1cf9b574e/sugar.go#L378) instead of fmt.Sprint. Does this make sense?

Reply:
That's a low price for keeping the code more readable imo, but it's not that big a deal. Feel free to ignore this comment if you disagree.